### PR TITLE
Stopping the automatic lower casing of the RestAPI url.

### PR DESCRIPTION
**Pull request by dillrye**
I have a Wordpress site that I am doing some work on for a customer which has capitals in the url path.  This change was needed to be able to use the api otherwise it would throw a 404.

### CODE REVIEW

File: RestAPI.cs [hunk 1]
```diff
@@ -43,12 +43,12 @@ public class RestAPI
             if (string.IsNullOrEmpty(url))
                 throw new Exception("Please use a valid WooCommerce Restful API url.");
 
-            url = url.Trim().ToLower().TrimEnd('/');
```
Comment (XiaoFaye) [inline RestAPI.cs]: was this not working?

Comment (dillrye) [inline RestAPI.cs]: I had a site like www.somesite.com/ABC/ and it would lowercase the url would return a 404. I tested this using postman as well.

Comment (XiaoFaye) [inline RestAPI.cs]: `url = url.Trim().ToLower().TrimEnd('/');`

This line already has ToLower(), what happened here?

**Comment by dillrye** [comment]
I made it so the actual url is not to lowered, only the checks.  for the site I am working on, an all lowercase url path does not work

File: RestAPI.cs (hunk 1 above)
Comment (dillrye) [inline RestAPI.cs]: the url i need is:  https://www.mywordpresssite/ABC/ internally the C# dll converts it to lower to this: https://www.mywordpresssite/abc/  which returns a 404 (Not found).  From looking at your code it looks like the ToLower() is so you can do some string compares, i created a separate variable so you can do the string compares without changing the actual url itself.

Comment (XiaoFaye) [inline RestAPI.cs]: Can you check your wordpress setting and see if you got ABC or abc?

image

Comment (dillrye) [inline RestAPI.cs]: Yes wordpress settings in both the URLs are ABC (upper case)

Comment (XiaoFaye) [inline RestAPI.cs]: If upper case is not necessary, change to abc should fix the issue.

Comment (dillrye) [inline RestAPI.cs]: This is a customer's site.  What is wrong with fixing the library?

Comment (XiaoFaye) [inline RestAPI.cs]: At least you can fix it by now, instead of waiting for the new version. The library will be updated.

Comment (dillrye) [inline RestAPI.cs]: Oh, this is a pull request, i did fix the issue and added the code on github.  Requesting you merge it into your main branch.

**Pull request merged**